src/scheduler/node_partition.c

```diff
@@ -965,6 +965,17 @@ resresv_can_fit_nodepart(status *policy, node_partition *np, resource_resv *resr
 
 	pass_flags = flags|UNSET_RES_ZERO;
 
+	/* Prelim check: node partition must have one or more nodes inside it */
+	if ((flags & COMPARE_TOTAL) && np->tot_nodes == 0) {
+		set_schd_error_codes(err, NEVER_RUN, NO_TOTAL_NODES);
+		if ((flags & RETURN_ALL_ERR)) {
+			can_fit = 0;
+			err->next = new_schd_error();
+			prev_err = err;
+			err = err->next;
+		} else
+			return 0;
+	}
```

Comment:
Make this part of check 1.  It's the COMPARE_TOTAL version of it.  I'd do an outer if() on flags & COMPARE_TOTAL.  Inside check for np->free_nodes or np->total_nodes based on if COMPARE_TOTAL is passed or not.